docs/src/man/basics.md

```diff
@@ -0,0 +1,1505 @@
+                                                  3 columns and 985 rows omitted
+```
+
+## Views
+
+You can simply create a `view` of a data frame (it is more efficient than creating a materialized selection).
+Here are the possible return value options.
+
+```jldoctest dataframe
+julia> german = copy(german_ref);
+
+julia> s_german1 = view(german, :, 2:5) # column subsetting
+1000×4 SubDataFrame
+  Row │ Age    Sex     Job    Housing
+      │ Int64  String  Int64  String
+──────┼───────────────────────────────
+    1 │    67  male        2  own
+    2 │    22  female      2  own
+    3 │    49  male        1  own
+    4 │    45  male        2  free
+    5 │    53  male        2  free
+    6 │    35  male        1  free
+    7 │    53  male        2  own
+    8 │    35  male        3  rent
+  ⋮   │   ⋮      ⋮       ⋮       ⋮
+  994 │    30  male        3  own
+  995 │    50  male        2  own
+  996 │    31  female      1  own
+  997 │    40  male        3  own
+  998 │    38  male        2  own
+  999 │    23  male        2  free
+ 1000 │    27  male        2  own
+                      985 rows omitted
+
+julia> s_german2 = @view german[end:-1:1, [1, 4]] # row and column subsetting
+1000×2 SubDataFrame
+  Row │ id     Job
+      │ Int64  Int64
+──────┼──────────────
+    1 │   999      2
+    2 │   998      2
+    3 │   997      2
+    4 │   996      3
+    5 │   995      1
+    6 │   994      2
+    7 │   993      3
+    8 │   992      1
+  ⋮   │   ⋮      ⋮
+  994 │     6      2
+  995 │     5      1
+  996 │     4      2
+  997 │     3      2
+  998 │     2      1
+  999 │     1      2
+ 1000 │     0      2
+     985 rows omitted
+```
+In the above example the number `2` and `5` represents the indices and as we know in `julia` indices start from `1`.
+So, from `2:5` it will access the column from `Age` to `Housing`.
+
+```jldoctest dataframe
+julia> @view german[1:5, 1]
+5-element view(::Vector{Int64}, 1:5) with eltype Int64:
+ 0
+ 1
+ 2
+ 3
+ 4
+
+julia> @view german[2, 2]
+0-dimensional view(::Vector{Int64}, 2) with eltype Int64:
+22
+
+julia> @view german[3, 2:5] # a DataFrameRow, the same as for german[3, 2:5] without a view
+DataFrameRow
+ Row │ Age    Sex     Job    Housing
+     │ Int64  String  Int64  String
+─────┼───────────────────────────────
+   3 │    49  male        1  own
+
+julia> @view german[2:5, 2:5] # here you can see creation of view is very fast
+4×4 SubDataFrame
+ Row │ Age    Sex     Job    Housing
+     │ Int64  String  Int64  String
+─────┼───────────────────────────────
+   1 │    22  female      2  own
+   2 │    49  male        1  own
+   3 │    45  male        2  free
+   4 │    53  male        2  free
+```
+
+In-place functions are safe to call, except when a view of the `DataFrame` (created via a `view`, `@view` or `*groupby*`) 
+or when a `DataFrame` created with `copycols=false` are in use.
+
+## Using `select`, and `select!`, `transform`, and `transform!`, `sort!`
```

Comment:
why `sort!` is in the same bag as the other functions?

Reply:
These operations I had to explain so I thought right now `sort!` also can be explained.

Reply:
explain `sort!` in a copletely separate secion (probably another PR after this one, as it is logically a completely different task)